src/components/AudioPlayer/helpers/useWaveSurfer.js

```diff
@@ -65,19 +65,22 @@ export default function useWaveSurfer(url) {
       wave.on('finish', () => {
         setPlayerConfig((prevConfig) => ({ ...prevConfig, isPlaying: false }))
       })
-
-      setWaveSurfer(wave)
+      window.wavesurfers = window.wavesurfers || []
+      window.wavesurfers.push(wave)
+      waveSurferRef.current = wave
     }
-
     return () => {
       wave?.stop()
       wave?.destroy()
-      setWaveSurfer(null)
+      const index = window.wavesurfers.indexOf(wave)
```

Comment:
Can we add a null check? Since window object is shared, we might not know if the array is cleared or has value. Some other instance might clear the array.

There might be cases which can cause the app to crash

```javascript
if (window.wavesurfers?.length){
  const index = window.wavesurfers.indexOf(wave)
      if (index !== -1) {
        window.wavesurfers.splice(index, 1)
      }
}
```